jupyter_sphinx/execute.py

```diff
@@ -264,18 +264,12 @@ def write_notebook_output(notebook, output_dir, notebook_name, location=None):
     )
     # Write a script too.  Note that utf-8 is the de facto
     # standard encoding for notebooks. 
-    ext = notebook.metadata.get("language_info", {}).get("file_extension", None)
```

Comment:
What happens now if this key doesn't exist and/or the kernel does not exist or provide the correct information? Does nbconvert degrade gracefully?

Or should these lines be wrapped in a try/except+logger.warning, to ensure this "non-essential" output does not kill the whole sphinx build if it errors?

Reply:
> Does nbconvert degrade gracefully?

Yes. Both if `language_info` is missing and if `language_info = {'name': 'unknown'}`. Annoyingly nbconvert ignores the extension even if it is stored in the metadata. Not sure if that occurs in the wild, and I'm inclined to make it an issue in nbconvert.

Reply:
I was wrong, nbconvert properly treats all metadata.

Reply:
Perhaps also test it against a non-standard kernel? Like in https://github.com/executablebooks/MyST-NB I now build a docs page using the https://github.com/evhub/coconut kernel:

```bash
pip install coconut
pip install ipython<=7.11.0  # required by coconut 1.4.3
coconut --jupyter # register kernel
```

Reply:
I decided to go with bash kernel.